pallets/parachain-staking/src/lib.rs

```diff
@@ -1650,6 +1671,47 @@ pub mod pallet {
 			<SelectedCandidates<T>>::put(collators);
 			(collator_count, delegation_count, total)
 		}
+
+		/// Apply the delegator intent for revoke and decrease, to build the
+		/// effective list of delegator, with their intended bond amount.
+		/// This is used later to compute the rewards.
+		fn get_rewardable_delegators(
+			collator: &T::AccountId,
+		) -> Vec<Bond<T::AccountId, BalanceOf<T>>>
+		where
+			T: Config,
+		{
+			<TopDelegations<T>>::get(collator)
+				.expect("all members of CandidateQ must be candidates")
+				.delegations
+				.into_iter()
+				.filter_map(|mut bond| {
+					let delegator = <DelegatorState<T>>::get(&bond.owner)
+						.expect("delegator state must be present for a bonded delegation");
+
+					match delegator.requests.requests.get(collator) {
+						// no revoke request, do nothing
+						None => Some(bond),
+
+						// Revoke request, filter the delegator out
+						Some(DelegationRequest {
+							action: DelegationChange::Revoke,
+							..
+						}) => None,
```

Comment:
maybe it's better to keep the delegator with amount zero to still have the same number of delegators, what do you think @4meta5 ?

Reply:
Although I cannot think of any reason to keep snapshot.delegations.len() == top_delegations.len(), but maybe it could be safer for some debugging reason.

We could also instead emit an event for this branch. This could be more efficient than a zero payout to the delegation which I think still has some cost?

Reply:
We definitely need some indication of when this branch is hit or people will be really confused and it'll be hard to debug for us.

Reply:
the null payment is already optimized by the pallet balance, it is not a problem. I don't think we need an additional event.

Reply:
ok then I agree, we should at least change it to a bond with amount zero cc @nbaztec

Reply:
Can we add logs instead? There are already `DelegationRevocationScheduled` and `DelegationDecreaseScheduled` events that will help to know what should happen

Reply:
Adding logs would work

Reply:
I think the generic trait `Balance` doesn't implement any concrete `zero` method but only `Default` method. While I'm pretty sure the default is/should be zero - is it safe enough to use that?

Reply:
FRAME provide trait bound for `Zero` on `Balance`